pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -854,7 +854,7 @@ func (m *kubeGenericRuntimeManager) doBackOff(pod *v1.Pod, container *v1.Contain
 	key := getStableKey(pod, container)
 	if backOff.IsInBackOffSince(key, ts) {
 		if ref, err := kubecontainer.GenerateContainerRef(pod, container); err == nil {
-			m.recorder.Eventf(ref, v1.EventTypeWarning, events.BackOffStartContainer, "Back-off restarting failed container")
+			m.recorder.Eventf(ref, v1.EventTypeWarning, events.BackOffStartContainer, "Back-off restarting failed container %s", container.Name)
```

Comment:
/hold

This increases the back off event writes per pod … the Kubelet throttles event writes and simply omits events over a certain threshold, so increasing the number and variety of events this way could easily drop any events fro other pods

Reply:
Is this actually the case?  The ref provided looks like it is already at the container granularity, so i'm not certain this actually increases the number or variety of events.  But then again, i'm not familiar with the deduplication logic.

Reply:
Fast code search results. It looks like the max events count is calculated based on localKey:

https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/client-go/tools/record/events_cache.go#L266-L268

Local key is the message (second return value in here):

https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/staging/src/k8s.io/client-go/tools/record/events_cache.go#L156-L173

So this change may in fact make everything worse - more events may be lost.

Given that you can get the name of failed container with `describe` command, closing this off for now.

/close